File: _includes/code/java-v6/src/test/java/ConfigureRQTest.java
import io.weaviate.client6.v1.api.WeaviateClient;
import io.weaviate.client6.v1.api.collections.CollectionHandle;
import io.weaviate.client6.v1.api.collections.Property;
import io.weaviate.client6.v1.api.collections.Quantization;
import io.weaviate.client6.v1.api.collections.VectorConfig;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.Map;

class ConfigureRQTest {

  private static WeaviateClient client;

  @BeforeAll
  public static void beforeAll() {
    // START ConnectCode
    String openaiApiKey = System.getenv("OPENAI_API_KEY");
    client = WeaviateClient.connectToLocal(
        config -> config.setHeaders(Map.of("X-OpenAI-Api-Key", openaiApiKey)));
    // END ConnectCode
  }

  @AfterAll
  public static void afterAll() throws Exception {
    client.close();
  }

  @Test
  void testEnableRQ() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }

    // START EnableRQ
    client.collections.create(
        "MyCollection",
        col -> col
            .vectorConfig(VectorConfig.text2vecContextionary(
                vc -> vc
                    // highlight-start
                    .quantization(Quantization.rq())
            // highlight-end
            ))
            .properties(Property.text("title")));
    // END EnableRQ
  }

  @Test
  void test1BitEnableRQ() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }

    // START 1BitEnableRQ
    client.collections.create(
        "MyCollection",
        col -> col
            .vectorConfig(VectorConfig.text2vecContextionary(
                vc -> vc
                    // highlight-start
                    .quantization(Quantization.rq(q -> q.bits(1)))
            // highlight-end
            ))
            .properties(Property.text("title")));
    // END 1BitEnableRQ
  }

  @Test
  void testUncompressed() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }

    // START Uncompressed
    client.collections.create(
        "MyCollection",
        col -> col
            .vectorConfig(VectorConfig.text2vecContextionary(
                vc -> vc
                    // highlight-start
                    .quantization(Quantization.uncompressed())
            // highlight-end
            ))
            .properties(Property.text("title")));
    // END Uncompressed
  }

  @Test
  void testRQWithOptions() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }

    // START RQWithOptions
    client.collections.create(
        "MyCollection",
        col -> col
            .vectorConfig(VectorConfig.text2vecContextionary(
                vc -> vc
                    // highlight-start
                    .quantization(Quantization.rq(q -> q
                        .bits(8) // Optional: Number of bits
                        .rescoreLimit(20) // Optional: Number of candidates to fetch before rescoring
                    ))
            // highlight-end
            ))
            .properties(Property.text("title")));
    // END RQWithOptions
  }

  // TODO[g-despot] Errors on collection update: TYPE_UPDATE_CLASS: bad request
  // :parse class update: invalid update for vector "default":
  // skipDefaultQuantization is immutable: attempted change from "true" to "false"
  @Test
  void testUpdateSchema() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }
    client.collections.create(collectionName, col -> col
        .vectorConfig(VectorConfig.text2vecContextionary(
            vc -> vc.quantization(Quantization.uncompressed())))
        .properties(Property.text("title")));

    // START UpdateSchema
    CollectionHandle<Map<String, Object>> collection = client.collections.use("MyCollection");
    collection.config.update(collectionName,
        c -> c.vectorConfig(VectorConfig.text2vecContextionary(vc -> vc.quantization(Quantization.rq()))));
    // END UpdateSchema
    // TODO[g-despot]: Verify the update
  }

  @Test
  void test1BitUpdateSchema() throws IOException {
    String collectionName = "MyCollection";
    if (client.collections.exists(collectionName)) {
      client.collections.delete(collectionName);
    }
    client.collections.create(collectionName, col -> col
        .vectorConfig(VectorConfig.text2vecContextionary(
            vc -> vc.quantization(Quantization.uncompressed())))
        .properties(Property.text("title")));

    // START 1BitUpdateSchema
    CollectionHandle<Map<String, Object>> collection = client.collections.use("MyCollection");
    collection.config.update(collectionName,
        c -> c
            .vectorConfig(VectorConfig.text2vecContextionary(vc -> vc.quantization(Quantization.rq(q -> q.bits(1))))));
    // END 1BitUpdateSchema
  }
}

File: _includes/code/csharp/ConfigureRQTest.cs
using System.Threading.Tasks;
using Weaviate.Client;
using Weaviate.Client.Models;
using Xunit;

namespace WeaviateProject.Tests;

public class ConfigureRQTest : IAsyncLifetime
{
    private WeaviateClient client;
    private const string COLLECTION_NAME = "MyCollection";

    // Runs before each test
    public async Task InitializeAsync()
    {
        // START ConnectCode
        client = await Connect.Local();
        // END ConnectCode

        // Clean slate for each test
        if (await client.Collections.Exists(COLLECTION_NAME))
        {
            await client.Collections.Delete(COLLECTION_NAME);
        }
    }

    // Runs after each test
    public Task DisposeAsync()
    {
        // No action needed here, as cleanup happens in InitializeAsync before the next test.
        return Task.CompletedTask;
    }

    [Fact]
    public async Task TestEnableRQ()
    {
        // START EnableRQ
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "MyCollection",
                Properties = [Property.Text("title")],
                VectorConfig = Configure.Vector(
                    "default",
                    v => v.Text2VecTransformers(),
                    index: new VectorIndex.HNSW
                    {
                        // highlight-start
                        Quantizer = new VectorIndex.Quantizers.RQ(),
                        // highlight-end
                    }
                ),
            }
        );
        // END EnableRQ
    }

    [Fact]
    public async Task Test1BitEnableRQ()
    {
        // START 1BitEnableRQ
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "MyCollection",
                Properties = [Property.Text("title")],
                VectorConfig = Configure.Vector(
                    "default",
                    v => v.Text2VecTransformers(),
                    index: new VectorIndex.HNSW
                    {
                        // highlight-start
                        Quantizer = new VectorIndex.Quantizers.RQ { Bits = 1 },
                        // highlight-end
                    }
                ),
            }
        );
        // END 1BitEnableRQ
    }

    [Fact]
    public async Task TestUncompressed()
    {
        // START Uncompressed
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "MyCollection",
                Properties = [Property.Text("title")],
                VectorConfig = Configure.Vector(
                    "default",
                    v => v.Text2VecTransformers(),
                    index: new VectorIndex.HNSW
                    {
                        // highlight-start
                        Quantizer = new VectorIndex.Quantizers.None { },
                        // highlight-end
                    }
                // highlight-end
                ),
            }
        );
        // END Uncompressed
    }

    [Fact]
    public async Task TestRQWithOptions()
    {
        // START RQWithOptions
        await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "MyCollection",
                Properties = [Property.Text("title")],
                VectorConfig = Configure.Vector(
                    "default",
                    v => v.Text2VecTransformers(),
                    index: new VectorIndex.HNSW
                    {
                        // highlight-start
                        Quantizer = new VectorIndex.Quantizers.RQ
                        {
                            Bits = 8, // Optional: Number of bits
                            RescoreLimit = 20, // Optional: Number of candidates to fetch before rescoring
                        },
                        // highlight-end
                    }
                ),
            }
        );
        // END RQWithOptions
    }

    [Fact]
    public async Task TestUpdateSchema()
    {
        // Note: Updating quantization settings on an existing collection is not supported by Weaviate
        // and will result in an error, as noted in the Java test. This test demonstrates the syntax for attempting the update.
        var collection = await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "MyCollection",
                Properties = [Property.Text("title")],
                VectorConfig = Configure.Vector("default", v => v.Text2VecTransformers()),
            }
        );

        // START UpdateSchema
        await collection.Config.Update(c =>
        {
            var vectorConfig = c.VectorConfig["default"];
            vectorConfig.VectorIndexConfig.UpdateHNSW(h =>
                h.Quantizer = new VectorIndex.Quantizers.RQ()
            );
        });
        // END UpdateSchema
    }

    [Fact]
    public async Task Test1BitUpdateSchema()
    {
        var collection = await client.Collections.Create(
            new CollectionCreateParams
            {
                Name = "MyCollection",
                Properties = [Property.Text("title")],
                VectorConfig = Configure.Vector("default", v => v.Text2VecTransformers()),
            }
        );

        // START 1BitUpdateSchema
        await collection.Config.Update(c =>
        {
            var vectorConfig = c.VectorConfig["default"];
            vectorConfig.VectorIndexConfig.UpdateHNSW(h =>
                h.Quantizer = new VectorIndex.Quantizers.RQ { Bits = 1 }
            );
        });
        // END 1BitUpdateSchema
    }
}
